src/configuration/config.c

```diff
@@ -615,7 +623,7 @@ bool Config_Option_set(Config_Option_Field field, const char *val) {
 		case Config_TIMEOUT:
 			{
 				long long timeout;
-				if(!_Config_ParsePositiveInteger(val, &timeout)) return false;
+				if(!_Config_ParseNonNegativeInteger(val, &timeout)) return false;
 				Config_timeout_set(timeout);
 			}
 			break;
```

Comment:
`Config_RESULTSET_MAX_SIZE` should use `_Config_ParsePositiveInteger`
`Config_VKEY_MAX_ENTITY_COUNT` should use `_Config_ParseNonNegativeInteger`

Reply:
`Config_RESULTSET_MAX_SIZE` defaults to unlimited when given a negative value, so I think `_Config_ParseInteger` makes sense for it.